File: 백준/Bronze/27891. 특별한 학교 이름 암호화/특별한 학교 이름 암호화.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int t;
    string name, formatn;
    vector<string> sname = {"branksomeh", "koreainter", "stjohnsbur", "northlondo"};
    vector<string> abbrev = {"BHA", "KIS", "SJA", "NLCS"};
    cin >> name;
    for (int i = 0; i < 4; i++) {
        t = name[0] - sname[i][0], formatn = "";
        if (t >= 0) {
            for (int j = 0; j < 10; j++)
                formatn += sname[i][j] + t > 122 ? char(sname[i][j] + t - 26) : char(sname[i][j] + t);
        } else {
            for (int j = 0; j < 10; j++)
                formatn += sname[i][j] + t < 97 ? char(sname[i][j] + t + 26) : char(sname[i][j] + t);
        }
        if (name == formatn) {
            cout << abbrev[i];
            break;
        }
    }
    return 0;
}

File: 백준/Bronze/27891. 특별한 학교 이름 암호화/특별한 학교 이름 암호화.py
import sys
input=lambda:sys.stdin.readline().rstrip()
name=input()
sname=["branksomeh","koreainter","stjohnsbur","northlondo"]
abbrev=["BHA","KIS","SJA","NLCS"]
for i in range(4):
    t,formatn=ord(name[0])-ord(sname[i][0]),""
    if t>=0:
        for c in sname[i]:
            formatn+=chr(ord(c)+t-26 if ord(c)+t>122 else ord(c)+t)
    else:
        for c in sname[i]:
            formatn+=chr(ord(c)+26+t if ord(c)+t<97 else ord(c)+t)
    if formatn==name:
        print(abbrev[i])
        break